Src/BootCamp.Chapter/FullCommand.cs

```diff
@@ -0,0 +1,32 @@
+﻿using System;
+using System.IO;
+using System.Linq;
+
+namespace BootCamp.Chapter
+{
+    internal class FullCommand
+    {
+        public FullCommand()
+        {
+        }
+
+        public EventHandler<FilterCompletedEventArgs> FilterCompleted;
+
+        internal void ExecuteCommand(Stream stream)
+        {
+            using var transactionStream = new TransactionStream(stream);
+            var transactions = transactionStream.ReadTransactionUntilEnd().GroupBy(x => x.ShopName).ToArray(); 
+
+            foreach (var shop in transactions)
+            {
+                OnFilterCompleted(shop);
+            }
+        }
+
+        private void OnFilterCompleted(IGrouping<string, Transaction> shop)
```

Comment:
``OnFilterCompleted`` is something that should handle the event, not call it. Therefore the ``On`` prefix should be amended.

And then it reads nicely: ``CompleteFilter``. What happens when filter is completed:
event is fired.